setup.py

```diff
@@ -19,8 +19,22 @@
 from setuptools import find_packages, setup
 from setuptools.command.test import test as TestCommand
 from version import version
+from pkg_resources import parse_version
 
-toil_version = '3.1.4'
+# Toil version check -- Raise warning instead of using intall_requires to avoid virtualenv conflicts
+toil_minor_version = 3.1
```

Comment:
Using floats for discrete math is suicide. It may work in this very particular case, since format() is truncating the tiny imprecisions but it is a really bad programming practice that I don't want to see anywhere in our code.

http://stackoverflow.com/questions/3730019/why-not-use-double-or-float-to-represent-currency/3730040#3730040

If you use ipython in a venv, you get the old non-setuptools version of parse_version, the one that returns a dumb tuple. Either don't use ipython for your interactive tests, or use the technique explained [here](https://coderwall.com/p/xdox9a/running-ipython-cleanly-inside-a-virtualenv) to launch ipython inside a venv:

```
(bd2k:/__me__/)hannes-mbpr:~ hannes $ virtualenv venv
New python executable in venv/bin/python
Installing setuptools, pip, wheel...done.
(bd2k:/__me__/)hannes-mbpr:~ hannes $ . venv/bin/activate
(venv)(bd2k:/__me__/)hannes-mbpr:~ hannes $ pip install ipython
… snip …
(venv)(bd2k:/__me__/)hannes-mbpr:~ hannes $ python -c 'import IPython; IPython.terminal.ipapp.launch_new_instance()'
Python 2.7.10 (default, Oct 23 2015, 19:19:21) 
In [1]: from pkg_resources import parse_version

In [2]: parse_version('1.0')
Out[2]: <Version('1.0')>
```

Note how it's not a tuple anymore.